src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/DebugFuncEval.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Internal.Runtime.CallInterceptor;
+
+namespace System.Runtime.InteropServices
+{
+    [McgIntrinsics]
+    internal static class AddrofIntrinsics
+    {
+        // This method is implemented elsewhere in the toolchain
+        internal static IntPtr AddrOf<T>(T ftn) { throw new PlatformNotSupportedException(); }
+    }
+}
+
+namespace Internal.Runtime.TypeLoader
+{
+    internal class DebugFuncEval
+    {
+        private static void HighLevelDebugFuncEvalHelperWithVariables(ref int param, ref LocalVariableSet arguments)
+        {
+            // Hard coding the argument integer here!
+            arguments.SetVar<int>(1, param);
+
+            // Obtain the target method address from the runtime
+            IntPtr targetAddress = RuntimeImports.RhpGetFuncEvalTargetAddress();
+
+            // Hard coding a single void return here
+            LocalVariableType[] returnAndArgumentTypes = new LocalVariableType[2];
+            returnAndArgumentTypes[0] = new LocalVariableType(typeof(void).TypeHandle, false, false);
+            returnAndArgumentTypes[1] = new LocalVariableType(typeof(int).TypeHandle, false, false);
+
+            // Hard coding static here
+            DynamicCallSignature dynamicCallSignature = new DynamicCallSignature(Internal.Runtime.CallConverter.CallingConvention.ManagedStatic, returnAndArgumentTypes, returnAndArgumentTypes.Length);
+
+            // Invoke the target method
+            Internal.Runtime.CallInterceptor.CallInterceptor.MakeDynamicCall(targetAddress, dynamicCallSignature, arguments);
+
+            // Signal to the debugger the func eval completes
+            IntPtr funcEvalCompleteCommandPointer;
+            unsafe
+            {
+                FuncEvalCompleteCommand* funcEvalCompleteCommand = stackalloc FuncEvalCompleteCommand[1];
```

Comment:
Oh, when I wrote "pointer to a stack allocated struct", what I meant is this:

```csharp
FuncEvalCompleteCommand funcEvalCompleteCommand = new FuncEvalCompleteCommand() { commandCode = 0 };
RuntimeImports.RhpSendCustomEventToDebugger(&funcEvalCompleteCommand, Unsafe.SizeOf<FuncEvalCompleteCommand>()); 
```

You can take the address of a local in C# without pinning or anything (that's what I meant by stack allocated struct - no need to `stackalloc` a 1-element array).